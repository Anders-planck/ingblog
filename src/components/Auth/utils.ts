import { z } from 'zod';

export const AuthFormSchema = z.object({
  email: z
    .string()
    .trim()
    .min(1, { message: 'Email required!' })
    .email({ message: 'Invalid email!' }),
  password: z
    .string()
    .trim()
    .min(1, { message: 'Password required!' })
    .min(8, { message: 'Password must have at least 8 characters!' }),
  name: z.string().trim().min(1, { message: 'Name required!' }),
  terms: z.boolean().refine((value) => value, { message: 'Accept terms!' }),
});

export const getAuthSchema = (type: 'login' | 'register') =>
  type === 'login' ? AuthFormSchema.pick({ email: true, password: true }) : AuthFormSchema;

export type AuthFormValues<T extends 'login' | 'register'> = T extends 'login'
  ? {
      email: string;
      password: string;
    }
  : {
      email: string;
      name: string;
      password: string;
      terms: boolean;
    };
