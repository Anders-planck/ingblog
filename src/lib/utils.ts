import { formatDistanceToNow } from 'date-fns';

export function formatPostDate(date: Date | number): string {
  const result = formatDistanceToNow(date, { addSuffix: true });
  return result.replace('about', 'posted').replace('less than', 'posted');
}
