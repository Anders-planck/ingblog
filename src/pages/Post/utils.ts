import { Post as PostType, Comment } from '@/types/post';

export const formatPost = (post: any) =>
  ({
    ...post,
    author: {
      name: post.profiles.full_name,
      avatar: post.profiles.avatar_url,
    },
    createdAt: new Date(post.created_at).toISOString(),
    likes: post.likes,
    bookmarks: post.bookmarks,
    comments: post.comments.map(formatComment),
    userLiked: post.likes.some((like: any) => like.author_id === post.author_id),
    userBookmarked: post.bookmarks.some((bookmark: any) => bookmark.author_id === post.author_id),
  }) as PostType;

export const formatComment = (comment: any) =>
  ({
    ...comment,
    author: {
      name: comment.profiles.full_name,
      avatar: comment.profiles.avatar_url,
      id: comment.author_id,
    },
  }) as Comment;
