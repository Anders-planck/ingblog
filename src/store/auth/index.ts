import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import { Session } from '@supabase/supabase-js';
import { RootState } from '@/store';

interface AuthState {
  user: any | null;
  session: Session | null;
  loginAttempted: boolean;
  profile: {
    full_name: string;
    phone: string;
    work: string;
    updated_at: Date;
  } | null;
}

const initialState: AuthState = {
  user: null,
  session: null,
  loginAttempted: false,
  profile: null,
};

const authSlice = createSlice({
  name: 'auth',
  initialState,
  reducers: {
    setUser: (state, action: PayloadAction<null | any>) => {
      state.user = action.payload;
    },
    setSession: (state, action: PayloadAction<null | Session>) => {
      state.session = action.payload;
    },
    setLoginAttempted: (state, action: PayloadAction<boolean>) => {
      state.loginAttempted = action.payload;
    },
    setProfile: (state, action: PayloadAction<null | AuthState['profile']>) => {
      state.profile = action.payload;
    },
  },
});

export const { setUser, setSession, setLoginAttempted, setProfile } = authSlice.actions;

export const { reducer } = authSlice;

export const selectIsAuthenticated = ({ auth }: RootState) => auth.user !== null;

export const selectUser = ({ auth }: RootState) => auth.user;

export const selectSession = ({ auth }: RootState) => auth.session;

export const selectProfile = ({ auth }: RootState) => auth.profile;

export const selectLoginAttempted = ({ auth }: RootState) => auth.loginAttempted;
