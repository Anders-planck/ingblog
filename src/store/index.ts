import { Action, configureStore, ThunkAction } from '@reduxjs/toolkit';
import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux';
import { reducer as searchReducer } from './app/search';
import { listenerMiddleware } from './listener';
import { reducer as authReducer } from './auth';

const makeStore = () =>
  configureStore({
    reducer: {
      auth: authReducer,
      search: searchReducer,
    },
    middleware: (getDefaultMiddlewares) =>
      getDefaultMiddlewares().concat(listenerMiddleware.middleware),
  });

const store = makeStore();
export default store;

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;

// Use throughout your app instead of plain `useDispatch` and `useSelector`
export const useAppDispatch = () => useDispatch<AppDispatch>();
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;
