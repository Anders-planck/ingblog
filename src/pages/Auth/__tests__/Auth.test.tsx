import { waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Authenticator from '@/pages/Auth';
import { renderWithProviders } from '@/tests/utils';
import { useAppSelector } from '@/store';
import { selectIsAuthenticated } from '@/store/auth';

jest.setTimeout(300000);

const mockNavigate = jest.fn().mockImplementation(() => {});
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockNavigate,
}));

jest.mock('@/lib/supabase', () => ({
  ...jest.requireActual('@/lib/supabase'),
  supabase: {
    ...jest.requireActual('@/lib/supabase').supabase,
    auth: {
      signInWithOAuth: jest.fn().mockResolvedValue({
        error: null,
        data: {},
      }),
      signInWithOtp: jest.fn().mockResolvedValue({
        error: null,
        data: {},
      }),
    },
  },
}));

jest.mock('@/store/listener', () => ({
  listenerMiddleware: {
    ...jest.requireActual('@/store/listener').listenerMiddleware,
  },
}));

jest.mock('@/store', () => ({
  ...jest.requireActual('@/store'),
  useAppSelector: jest.fn(),
}));

afterAll(() => {
  jest.restoreAllMocks();
});

test('renders Authenticator', async () => {
  let containerMaster: any;

  await waitFor(() => {
    const { container } = renderWithProviders(<Authenticator />);
    containerMaster = container;
  });
  expect(containerMaster).toMatchSnapshot();
  expect(containerMaster).toHaveTextContent("Welcome to Isa-blog, Let's get started!");
});

const navigate = jest.fn();
jest.spyOn(require('react-router-dom'), 'useNavigate').mockReturnValue(navigate);

test('navigates to home route if user is authenticated', async () => {
  (useAppSelector as jest.Mock).mockImplementation(
    (selector) => selector === selectIsAuthenticated
  );

  await waitFor(() => {
    renderWithProviders(<Authenticator />);
  });

  expect(navigate).toHaveBeenCalledWith('/');
});

const signInWithOAuth = jest.fn().mockResolvedValue({ error: null, data: {} });
jest
  .spyOn(require('@/lib/supabase').supabase.auth, 'signInWithOAuth')
  .mockImplementation(signInWithOAuth);

const signInWithOtp = jest.fn().mockResolvedValue({ error: null, data: {} });
jest
  .spyOn(require('@/lib/supabase').supabase.auth, 'signInWithOtp')
  .mockImplementation(signInWithOtp);

test('does not navigate to home route if user is not authenticated', async () => {
  (useAppSelector as jest.Mock).mockImplementation(
    (selector) => selector !== selectIsAuthenticated
  );

  await waitFor(() => {
    renderWithProviders(<Authenticator />);
  });

  expect(navigate).not.toHaveBeenCalled();
});

test('check if buttons are on the screen', async () => {
  let containerMaster: any;
  let getByTestIdMaster: any;
  await waitFor(() => {
    const { container, getByTestId } = renderWithProviders(<Authenticator />);
    containerMaster = container;
    getByTestIdMaster = getByTestId;
  });
  expect(containerMaster).toMatchSnapshot();

  const githubButton = getByTestIdMaster('github-button');
  expect(githubButton).toBeInTheDocument();

  await waitFor(() => {
    githubButton.click();
    expect(signInWithOAuth).toHaveBeenCalledWith({ provider: 'github' });
  });

  const googleButton = getByTestIdMaster('google-button');
  expect(googleButton).toBeInTheDocument();

  await waitFor(() => {
    googleButton.click();
    expect(signInWithOAuth).toHaveBeenCalledWith({ provider: 'google' });
  });

  const emailInput = getByTestIdMaster('email-input');
  expect(emailInput).toBeInTheDocument();
  await userEvent.type(emailInput, 'john@doe.com');
  expect(emailInput).toHaveValue('john@doe.com');

  const button = getByTestIdMaster('login-button');
  expect(button).toBeInTheDocument();

  await waitFor(() => {
    button.click();
    expect(signInWithOtp).toHaveBeenCalledWith({ email: 'john@doe.com' });
  });
});
