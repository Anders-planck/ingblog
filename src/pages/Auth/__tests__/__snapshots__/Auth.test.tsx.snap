// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check if buttons are on the screen 1`] = `
<div>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <div>
    <div
      class="m_7485cace mantine-Container-root"
      data-size="xs"
      style="--container-size: var(--container-size-xs);"
    >
      <div
        class="m_8bffd616 mantine-Flex-root __m__-r26"
        style="height: calc(100vh - 100px); align-items: center; justify-content: center; flex-direction: column;"
      >
        <div
          class="m_1b7284a3 mantine-Paper-root"
          data-with-border="true"
          style="--paper-radius: var(--mantine-radius-md); width: 100%;"
        >
          <p
            class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
            data-size="lg"
            style="--text-fz: var(--mantine-font-size-lg); --text-lh: var(--mantine-line-height-lg); font-weight: 500;"
          >
            Welcome to Isa-blog, Let's get started!
          </p>
          <div
            class="m_4081bf90 mantine-Group-root"
            data-grow="true"
            style="--group-child-width: calc(50% - (var(--mantine-spacing-md) - var(--mantine-spacing-md) / 2)); --group-gap: var(--mantine-spacing-md); --group-align: center; --group-justify: flex-start; --group-wrap: wrap;"
          >
            <button
              class="mantine-focus-auto mantine-active m_77c9d27d mantine-Button-root m_87cf2631 mantine-UnstyledButton-root"
              data-block="true"
              data-testid="google-button"
              data-variant="default"
              data-with-left-section="true"
              style="--button-radius: var(--mantine-radius-xl); --button-bg: var(--mantine-color-default); --button-hover: var(--mantine-color-default-hover); --button-color: var(--mantine-color-default-color); --button-bd: calc(0.0625rem * var(--mantine-scale)) solid var(--mantine-color-default-border);"
              type="button"
            >
              <span
                class="m_80f1301b mantine-Button-inner"
              >
                <span
                  class="m_a74036a mantine-Button-section"
                  data-position="left"
                >
                  <svg
                    preserveAspectRatio="xMidYMid"
                    style="width: 0.9rem; height: 0.9rem;"
                    viewBox="0 0 256 262"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M255.878 133.451c0-10.734-.871-18.567-2.756-26.69H130.55v48.448h71.947c-1.45 12.04-9.283 30.172-26.69 42.356l-.244 1.622 38.755 30.023 2.685.268c24.659-22.774 38.875-56.282 38.875-96.027"
                      fill="#4285F4"
                    />
                    <path
                      d="M130.55 261.1c35.248 0 64.839-11.605 86.453-31.622l-41.196-31.913c-11.024 7.688-25.82 13.055-45.257 13.055-34.523 0-63.824-22.773-74.269-54.25l-1.531.13-40.298 31.187-.527 1.465C35.393 231.798 79.49 261.1 130.55 261.1"
                      fill="#34A853"
                    />
                    <path
                      d="M56.281 156.37c-2.756-8.123-4.351-16.827-4.351-25.82 0-8.994 1.595-17.697 4.206-25.82l-.073-1.73L15.26 71.312l-1.335.635C5.077 89.644 0 109.517 0 130.55s5.077 40.905 13.925 58.602l42.356-32.782"
                      fill="#FBBC05"
                    />
                    <path
                      d="M130.55 50.479c24.514 0 41.05 10.589 50.479 19.438l36.844-35.974C195.245 12.91 165.798 0 130.55 0 79.49 0 35.393 29.301 13.925 71.947l42.211 32.783c10.59-31.477 39.891-54.251 74.414-54.251"
                      fill="#EB4335"
                    />
                  </svg>
                </span>
                <span
                  class="m_811560b9 mantine-Button-label"
                >
                  Google
                </span>
              </span>
            </button>
            <button
              class="mantine-focus-auto mantine-active m_77c9d27d mantine-Button-root m_87cf2631 mantine-UnstyledButton-root"
              data-block="true"
              data-testid="github-button"
              data-variant="default"
              data-with-left-section="true"
              style="--button-radius: var(--mantine-radius-xl); --button-bg: var(--mantine-color-default); --button-hover: var(--mantine-color-default-hover); --button-color: var(--mantine-color-default-color); --button-bd: calc(0.0625rem * var(--mantine-scale)) solid var(--mantine-color-default-border);"
              type="button"
            >
              <span
                class="m_80f1301b mantine-Button-inner"
              >
                <span
                  class="m_a74036a mantine-Button-section"
                  data-position="left"
                >
                  <svg
                    class="tabler-icon tabler-icon-brand-github "
                    fill="none"
                    height="24"
                    stroke="#fff"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    style="width: 1rem; height: 1rem;"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M9 19c-4.3 1.4 -4.3 -2.5 -6 -3m12 5v-3.5c0 -1 .1 -1.4 -.5 -2c2.8 -.3 5.5 -1.4 5.5 -6a4.6 4.6 0 0 0 -1.3 -3.2a4.2 4.2 0 0 0 -.1 -3.2s-1.1 -.3 -3.5 1.3a12.3 12.3 0 0 0 -6.2 0c-2.4 -1.6 -3.5 -1.3 -3.5 -1.3a4.2 4.2 0 0 0 -.1 3.2a4.6 4.6 0 0 0 -1.3 3.2c0 4.6 2.7 5.7 5.5 6c-.6 .6 -.6 1.2 -.5 2v3.5"
                    />
                  </svg>
                </span>
                <span
                  class="m_811560b9 mantine-Button-label"
                >
                  Github
                </span>
              </span>
            </button>
          </div>
          <div
            class="m_3eebeb36 mantine-Divider-root"
            data-orientation="horizontal"
            data-with-label="true"
            role="separator"
            style="margin-block: var(--mantine-spacing-lg);"
          >
            <span
              class="m_9e365f20 mantine-Divider-label"
              data-position="center"
            >
              Or continue with email
            </span>
          </div>
          <div
            class="m_6d731127 mantine-Stack-root"
            style="--stack-gap: var(--mantine-spacing-md); --stack-align: stretch; --stack-justify: flex-start;"
          >
            <div
              class="m_46b77525 mantine-InputWrapper-root mantine-TextInput-root"
            >
              <label
                class="m_8fdc1311 mantine-InputWrapper-label mantine-TextInput-label"
                data-required="true"
                for="mantine-pwsfe2dwp"
                id="mantine-pwsfe2dwp-label"
              >
                Email
                <span
                  aria-hidden="true"
                  class="m_78a94662 mantine-InputWrapper-required mantine-TextInput-required"
                >
                   *
                </span>
              </label>
              <div
                class="m_6c018570 mantine-Input-wrapper mantine-TextInput-wrapper"
                data-variant="default"
                style="--input-radius: var(--mantine-radius-md);"
              >
                <input
                  aria-invalid="false"
                  class="m_8fb7ebe7 mantine-Input-input mantine-TextInput-input"
                  data-testid="email-input"
                  data-variant="default"
                  id="mantine-pwsfe2dwp"
                  placeholder="hello@mantine.dev"
                  required=""
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            class="m_4081bf90 mantine-Group-root"
            style="--group-gap: var(--mantine-spacing-md); --group-align: center; --group-justify: flex-end; --group-wrap: wrap;"
          >
            <button
              class="mantine-focus-auto mantine-active m_77c9d27d mantine-Button-root m_87cf2631 mantine-UnstyledButton-root"
              data-testid="login-button"
              style="--button-radius: var(--mantine-radius-xl); --button-color: var(--mantine-color-white);"
              type="button"
            >
              <span
                class="m_80f1301b mantine-Button-inner"
              >
                <span
                  class="m_811560b9 mantine-Button-label"
                >
                  Let's go
                </span>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders Authenticator 1`] = `
<div>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <div>
    <div
      class="m_7485cace mantine-Container-root"
      data-size="xs"
      style="--container-size: var(--container-size-xs);"
    >
      <div
        class="m_8bffd616 mantine-Flex-root __m__-r1"
        style="height: calc(100vh - 100px); align-items: center; justify-content: center; flex-direction: column;"
      >
        <div
          class="m_1b7284a3 mantine-Paper-root"
          data-with-border="true"
          style="--paper-radius: var(--mantine-radius-md); width: 100%;"
        >
          <p
            class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
            data-size="lg"
            style="--text-fz: var(--mantine-font-size-lg); --text-lh: var(--mantine-line-height-lg); font-weight: 500;"
          >
            Welcome to Isa-blog, Let's get started!
          </p>
          <div
            class="m_4081bf90 mantine-Group-root"
            data-grow="true"
            style="--group-child-width: calc(50% - (var(--mantine-spacing-md) - var(--mantine-spacing-md) / 2)); --group-gap: var(--mantine-spacing-md); --group-align: center; --group-justify: flex-start; --group-wrap: wrap;"
          >
            <button
              class="mantine-focus-auto mantine-active m_77c9d27d mantine-Button-root m_87cf2631 mantine-UnstyledButton-root"
              data-block="true"
              data-testid="google-button"
              data-variant="default"
              data-with-left-section="true"
              style="--button-radius: var(--mantine-radius-xl); --button-bg: var(--mantine-color-default); --button-hover: var(--mantine-color-default-hover); --button-color: var(--mantine-color-default-color); --button-bd: calc(0.0625rem * var(--mantine-scale)) solid var(--mantine-color-default-border);"
              type="button"
            >
              <span
                class="m_80f1301b mantine-Button-inner"
              >
                <span
                  class="m_a74036a mantine-Button-section"
                  data-position="left"
                >
                  <svg
                    preserveAspectRatio="xMidYMid"
                    style="width: 0.9rem; height: 0.9rem;"
                    viewBox="0 0 256 262"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M255.878 133.451c0-10.734-.871-18.567-2.756-26.69H130.55v48.448h71.947c-1.45 12.04-9.283 30.172-26.69 42.356l-.244 1.622 38.755 30.023 2.685.268c24.659-22.774 38.875-56.282 38.875-96.027"
                      fill="#4285F4"
                    />
                    <path
                      d="M130.55 261.1c35.248 0 64.839-11.605 86.453-31.622l-41.196-31.913c-11.024 7.688-25.82 13.055-45.257 13.055-34.523 0-63.824-22.773-74.269-54.25l-1.531.13-40.298 31.187-.527 1.465C35.393 231.798 79.49 261.1 130.55 261.1"
                      fill="#34A853"
                    />
                    <path
                      d="M56.281 156.37c-2.756-8.123-4.351-16.827-4.351-25.82 0-8.994 1.595-17.697 4.206-25.82l-.073-1.73L15.26 71.312l-1.335.635C5.077 89.644 0 109.517 0 130.55s5.077 40.905 13.925 58.602l42.356-32.782"
                      fill="#FBBC05"
                    />
                    <path
                      d="M130.55 50.479c24.514 0 41.05 10.589 50.479 19.438l36.844-35.974C195.245 12.91 165.798 0 130.55 0 79.49 0 35.393 29.301 13.925 71.947l42.211 32.783c10.59-31.477 39.891-54.251 74.414-54.251"
                      fill="#EB4335"
                    />
                  </svg>
                </span>
                <span
                  class="m_811560b9 mantine-Button-label"
                >
                  Google
                </span>
              </span>
            </button>
            <button
              class="mantine-focus-auto mantine-active m_77c9d27d mantine-Button-root m_87cf2631 mantine-UnstyledButton-root"
              data-block="true"
              data-testid="github-button"
              data-variant="default"
              data-with-left-section="true"
              style="--button-radius: var(--mantine-radius-xl); --button-bg: var(--mantine-color-default); --button-hover: var(--mantine-color-default-hover); --button-color: var(--mantine-color-default-color); --button-bd: calc(0.0625rem * var(--mantine-scale)) solid var(--mantine-color-default-border);"
              type="button"
            >
              <span
                class="m_80f1301b mantine-Button-inner"
              >
                <span
                  class="m_a74036a mantine-Button-section"
                  data-position="left"
                >
                  <svg
                    class="tabler-icon tabler-icon-brand-github "
                    fill="none"
                    height="24"
                    stroke="#fff"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    style="width: 1rem; height: 1rem;"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M9 19c-4.3 1.4 -4.3 -2.5 -6 -3m12 5v-3.5c0 -1 .1 -1.4 -.5 -2c2.8 -.3 5.5 -1.4 5.5 -6a4.6 4.6 0 0 0 -1.3 -3.2a4.2 4.2 0 0 0 -.1 -3.2s-1.1 -.3 -3.5 1.3a12.3 12.3 0 0 0 -6.2 0c-2.4 -1.6 -3.5 -1.3 -3.5 -1.3a4.2 4.2 0 0 0 -.1 3.2a4.6 4.6 0 0 0 -1.3 3.2c0 4.6 2.7 5.7 5.5 6c-.6 .6 -.6 1.2 -.5 2v3.5"
                    />
                  </svg>
                </span>
                <span
                  class="m_811560b9 mantine-Button-label"
                >
                  Github
                </span>
              </span>
            </button>
          </div>
          <div
            class="m_3eebeb36 mantine-Divider-root"
            data-orientation="horizontal"
            data-with-label="true"
            role="separator"
            style="margin-block: var(--mantine-spacing-lg);"
          >
            <span
              class="m_9e365f20 mantine-Divider-label"
              data-position="center"
            >
              Or continue with email
            </span>
          </div>
          <div
            class="m_6d731127 mantine-Stack-root"
            style="--stack-gap: var(--mantine-spacing-md); --stack-align: stretch; --stack-justify: flex-start;"
          >
            <div
              class="m_46b77525 mantine-InputWrapper-root mantine-TextInput-root"
            >
              <label
                class="m_8fdc1311 mantine-InputWrapper-label mantine-TextInput-label"
                data-required="true"
                for="mantine-d9oe3idk5"
                id="mantine-d9oe3idk5-label"
              >
                Email
                <span
                  aria-hidden="true"
                  class="m_78a94662 mantine-InputWrapper-required mantine-TextInput-required"
                >
                   *
                </span>
              </label>
              <div
                class="m_6c018570 mantine-Input-wrapper mantine-TextInput-wrapper"
                data-variant="default"
                style="--input-radius: var(--mantine-radius-md);"
              >
                <input
                  aria-invalid="false"
                  class="m_8fb7ebe7 mantine-Input-input mantine-TextInput-input"
                  data-testid="email-input"
                  data-variant="default"
                  id="mantine-d9oe3idk5"
                  placeholder="hello@mantine.dev"
                  required=""
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            class="m_4081bf90 mantine-Group-root"
            style="--group-gap: var(--mantine-spacing-md); --group-align: center; --group-justify: flex-end; --group-wrap: wrap;"
          >
            <button
              class="mantine-focus-auto mantine-active m_77c9d27d mantine-Button-root m_87cf2631 mantine-UnstyledButton-root"
              data-testid="login-button"
              style="--button-radius: var(--mantine-radius-xl); --button-color: var(--mantine-color-white);"
              type="button"
            >
              <span
                class="m_80f1301b mantine-Button-inner"
              >
                <span
                  class="m_811560b9 mantine-Button-label"
                >
                  Let's go
                </span>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
